### PR TITLE
BREAKING CHANGE: framework migration

### PR DESCRIPTION
**Pull request by isadoral**

### CODE REVIEW

**Comment by thatsddr** [comment]
The duplication that sonarcloud is complaining about can be fixed. E.g. for the [VOD source](https://sonarcloud.io/component_measures?metric=new_duplicated_lines_density&selected=spring-media_ott-tfprovider-awsmt%3Aawsmt%2Fdata_source_vod_source.go&view=list&branch=framework-migration&id=spring-media_ott-tfprovider-awsmt) and [Live source](https://sonarcloud.io/component_measures?metric=new_duplicated_lines_density&selected=spring-media_ott-tfprovider-awsmt%3Aawsmt%2Fdata_source_live_source.go&view=list&branch=framework-migration&id=spring-media_ott-tfprovider-awsmt) schemas you can define a map with all the common fields in one of the 2 files or a utils file. You can then import the map in the other file or both, depending on where you decide to put it, make a deep copy, add the missing files (vod_source_name and live_source_name, respectively), and then you can use it. You can do the same for the resource files.

File: examples/source_location.tf [hunk 1]
```diff
@@ -1,12 +1,19 @@
 resource "awsmt_source_location" "example" {
-  name = "example_source_location"
-  http_configuration_url = "https://ott-mediatailor-test.s3.eu-central-1.amazonaws.com/"
+  source_location_name = "examplesourcelocation"
+  http_configuration = {
+    hc_base_url = "https://ott-mediatailor-test.s3.eu-central-1.amazonaws.com"
```
Comment (thatsddr) [inline examples/source_location.tf:4]: what does `hc` mean in this context?

File: awsmt/resource_playback_configuration_test.go [hunk 2]
```diff
@@ -1,390 +1,137 @@
...
-func TestAccPlaybackConfigurationResourceBasic(t *testing.T) {
-	resourceName := "awsmt_playback_configuration.r1"
+func TestAccPlaybackConfigurationResource(t *testing.T) {
 	resource.Test(t, resource.TestCase{
-		PreCheck:          func() { testAccPreCheck(t) },
-		ProviderFactories: ProviderFactories,
-		CheckDestroy:      testAccCheckPlaybackConfigurationDestroy,
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
 		Steps: []resource.TestStep{
```
Comment (thatsddr) [inline awsmt/resource_playback_configuration_test.go:12]: I think resources need more testing, especially when it comes to updating them

**Review by thatsddr — CHANGES_REQUESTED** [review]
You can refer to the comments for the proposed changes. I know they may be unclear. Don't hesitate to reach out so I can explain each one of them

File: awsmt/resource_playback_configuration_test.go (hunk 2 above)
Comment (isadoral) [reply awsmt/resource_playback_configuration_test.go:12]: I tested updating everything I know how to change / what can be possible inputs for it, im not fully sure how to add more tests atm.

**Pull request merged**